### PR TITLE
Fix timestamp collection to not impact perf

### PR DESCRIPTION
**Pull request by CoderHam**
- Sacrfices true compute output time

### CODE REVIEW

File: src/libtorch.cc [hunk 1]
```diff
@@ -929,6 +907,15 @@ ModelInstanceState::ReadOutputTensors(
...
                                         : TRITONSERVER_MEMORY_GPU,
         (device_.type() == torch::kCPU) ? 0 : device_.index());
+
+    // Due to pytorch running inference asynchronously, we need set timestamp
+    // after first output is read to get a more accuracte estimate of compute
+    // infer time. This means the compute output time reported would be smaller
+    // than it is in reality. We do this because synchronizing manually causes a
+    // large drop in performance for fast running models.
+    if (idx == 0) {
+      SET_TIMESTAMP(*compute_end_ns);
```
Comment (GuanLuo) [inline src/libtorch.cc:917]: Shouldn't you place this after line 890?
https://github.com/triton-inference-server/pytorch_backend/pull/23/files#diff-859dcc1d94f2d13b0d19e7a14af430cd18eb66395a6f69f2d4886854319c4467L890

File: src/libtorch.cc [hunk 2]
```diff
@@ -929,6 +907,15 @@ ModelInstanceState::ReadOutputTensors(
         (device_.type() == torch::kCPU) ? TRITONSERVER_MEMORY_CPU
                                         : TRITONSERVER_MEMORY_GPU,
         (device_.type() == torch::kCPU) ? 0 : device_.index());
+
```
Comment (deadeyegoodwin) [inline src/libtorch.cc:910]: You should be able to move this to line 897 above just after output_buffer pointer is assigned.

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: src/libtorch.cc (hunk 2 above)
Comment (CoderHam) [reply src/libtorch.cc:910]: I did try that first. I shared number for the same in the ticket. It appears the synchronization occurs during the responder.ProcessTensor call.

Comment (GuanLuo) [reply src/libtorch.cc:910]: Why would that be the case? Are you saying that the content in the output_buffer is not valid until within the responder? But the responder doesn't interact with the torch library at all, how does the synchronization happen?

Comment (CoderHam) [reply src/libtorch.cc:910]: I am not 100% clear but this is what I believe is happening:
The output_buffer is a pointer to memory on GPU that the output tensor from pytorch will be stored. While trying to access data at the location, pytorch synchronizes and populates the output_buffer. While this is odd it does help with performance.
I could not find any documentation from pytorch that explains this behavior.

Comment (deadeyegoodwin) [reply src/libtorch.cc:910]: There's no way for pytorch to do that. Once we have a char* pointer to the data, accessing it via that pointer will not trigger anything that pytorch can be aware of (if there was some OS level page protection on the memory there could be some OS action triggered... but not in the pytorch code).

Comment (CoderHam) [reply src/libtorch.cc:910]: I can run some more tests to understand what causes this behavior.

**Review by deadeyegoodwin — APPROVED** [review]

**Review by GuanLuo — APPROVED** [review]

**Pull request merged**